File: pe/nonvisualstudio/clojure/Spec/Problem0006.Spec.clj
;(load-file "C:/Users/Thomas/Documents/GitHub/practice/pe/nonvisualstudio/clojure/Spec/Problem0006.Spec.clj")

(ns pe.clojure.implementation.problem0006.spec)

(load-file "C:/Users/Thomas/Documents/GitHub/practice/pe/nonvisualstudio/clojure/Implementation/Problem0006.clj")
(use 'pe.clojure.implementation.problem0006)
(use 'clojure.test)

(deftest Problem0006

  (testing "square"
    (is (= 25 (square 5)))
    (is (= 100 (square 10))))

  (testing "sumOfSquares"
    (is (= 5 (sumOfSquares 2)))
    (is (= 385 (sumOfSquares 10))))

  (testing "squareOfSums"
    (is (= 9 (squareOfSums 2)))
    (is (= 3025 (squareOfSums 10))))

  (testing "differenceBetweenSquareOfSumsAndSumOfSquares"
    (is (= 4 (differenceBetweenSquareOfSumsAndSumOfSquares 2)))
    (is (= 2640 (differenceBetweenSquareOfSumsAndSumOfSquares 10)))
    (is (= 25164150 (differenceBetweenSquareOfSumsAndSumOfSquares 100)))))


(run-tests)

File: pe/nonvisualstudio/haskell/Spec/Problem0006.Spec.hs
{-
ghci c:\Users\Thomas\Documents\GitHub\practice\pe\nonvisualstudio\haskell\Spec\Problem0006.Spec.hs
 c:\Users\Thomas\Documents\GitHub\practice\pe\nonvisualstudio\haskell\Implementation\Problem0006.hs
-}
-- :r :q :set +s for times

module Problem0006Tests where

  import Test.HUnit
  import System.IO

  import Problem0006

  testCases = TestList
    [
      TestCase $ assertEqual "squareOfSum 10 should return 3025." 3025 (squareOfSum 10),
      TestCase $ assertEqual "sumOfSquares 10 should return 385." 385 (sumOfSquares 10),
      TestCase $ assertEqual "differenceBetweenSumOfSquarsAndSquareOfSum 10 should return 2640." 2640 (differenceBetweenSumOfSquarsAndSquareOfSum 10),
      TestCase $ assertEqual "differenceBetweenSumOfSquarsAndSquareOfSum 100 should return 25164150." 25164150 (differenceBetweenSumOfSquarsAndSquareOfSum 100)
    ]

  tests = runTestTT testCases
